src/tests/ftest/dfuse/daos_build.py

```diff
@@ -208,6 +208,7 @@ def run_build_test(self, cache_mode, intercept=False, run_on_vms=False):
 
         cmds = ['python3 -m venv {}/venv'.format(mount_dir),
                 'git clone https://github.com/daos-stack/daos.git {}'.format(build_dir),
+                'git -C {} checkout release/2.4'.format(build_dir),
```

Comment:
We'll have the same problem when master branches to the next 2.x release.
If we wanted a more generic solution, we could rely on
```
$ daos version -j
{
  "response": {
    "name": "DAOS CLI",
    "version": "2.5.100",
    "revision": "0aae80f5909b8ec63f9b46dd31bb56cf88d23fa2",
    "dirty": true
  },
  "error": null,
  "status": 0
}
```
And either
1. Checkout the exact revision hash. Which means we would build the exact same commit that the PR/branch is running on.
2. Map the version to master, 2.4, etc.

If we do that, it should go into master as well, so no need to block this PR. And I could make that change myself. Thoughts?

Reply:
This has only come up as a problem now despite having branched in June, I agree that if/when we find we need to do this on release/2.6 then we should generalise it but for now I don't see it as being a major benefit.

Reply:
But prior to this change, the release/2.4 version of this test was building master, correct? So even if it wasn't failing, it wasn't quite doing the right thing

Reply:
There's no reason for it to match.  We could just as well put in a SHA hash here.